### PR TITLE
Support no resampling in TimeSeriesDataset

### PR DESCRIPTION
**Pull request by milesgranger**
Access the timeseries data without any forced resampling/aggregations

Moving over from #497

### CODE REVIEW

File: gordo_components/dataset/datasets.py [hunk 1]
```diff
@@ -86,42 +72,116 @@ def __init__(
...
+        """
+        Performs as :class:`RawTimeSeriesDataset`, but with additional
+        resampling functionality
+
+        A TimeSeriesDataset is a dataset backed by RawTimeSeriesDataset, but resampled.
+        Time series are still aligned, and (optionally) filtered as with
+        :class:`gordo_components.dataset.datasets.RawTimeSeriesDataset`
+
+        Parameters
+        ----------
```
Comment (epa095) [inline gordo_components/dataset/datasets.py]: It is a bit subpar that important arguments to TimeSeriesDataset are no longer documented directly in its init, but one has to chase them down in the superclass. Also, they are required, but that is no longer visible from the signature :-s
IDK if this is a argument for adding them here as explicit arguments, or for merging the two raw and non-raw timeseriesdatasets

File: tests/gordo_components/dataset/test_dataset.py [hunk 2]
```diff
@@ -173,21 +178,25 @@ def test_row_filter(self):
...
         X, _ = TimeSeriesDataset(
-            MockDataSource(),
-            start,
-            end,
+            data_provider=MockDataSource(),
+            from_ts=start,
+            to_ts=end,
             tag_list=tag_list,
             row_filter="'Tag 1' / 'Tag 3' < 0.999",
         ).get_data()
```
Comment (EinarJohnsen) [inline tests/gordo_components/dataset/test_dataset.py]: Not saying it should be done in this PR, but we should might think about using pytest to parameterize this instead, it looks like unnecessary duplication.

File: gordo_components/dataset/datasets.py [hunk 3]
```diff
@@ -98,13 +99,18 @@ def get_data(self) -> Tuple[pd.DataFrame, Optional[pd.DataFrame]]:
...
-        data: pd.DataFrame = self.join_timeseries(
-            series_iter,
-            self.from_ts,
-            self.to_ts,
-            self.resolution,
-            aggregation_methods=self.aggregation_methods,
-        )
+
+        # Resample if we have a resolution set, otherwise simply join the series.
+        if self.resolution is None:
```
Comment (epa095) [inline gordo_components/dataset/datasets.py]: Is it not more pythonic to do `if self.resolution`...`else`? Then empty string is also regarded as Falsy, just as `False` and `None`

File: tests/gordo_components/dataset/test_dataset.py [hunk 4]
```diff
@@ -238,6 +225,22 @@ def test_aggregation_methods(self):
...
+    kwargs = dict(
+        data_provider=MockDataSource(),
+        tag_list=[
+            SensorTag("Tag 1", None),
+            SensorTag("Tag 2", None),
+            SensorTag("Tag 3", None),
+        ],
+        from_ts=dateutil.parser.isoparse("2017-12-25 06:00:00Z"),
+        to_ts=dateutil.parser.isoparse("2017-12-29 06:00:00Z"),
+    )
```
Comment (ryanjdillon) [inline tests/gordo_components/dataset/test_dataset.py:238]: You could refactor this out and append test specific keys as needed.

**Review by ryanjdillon — APPROVED** [review]
LGTM :+1:

**Review by epa095 — APPROVED** [review]
LGTM

**Comment by milesgranger** [comment]
@choukha This is what you were asking about a while ago, access to raw data without any resampling. Will be out in the next release. Or use master of course. :+1:

**Pull request merged**